In [125]:
import pandas as pd
import numpy as np

# 2013-2014 Data

In [126]:
# 2013-2014
path = '/Users/carolinesklaver/Desktop/Capstone/NHANES/data/csv_data/2013-2014'

import os
os.chdir(path)

## PHQ-9 Survey - part of Questionnaire

In [127]:
DPQ_2013_2014 = pd.read_csv('./Questionnaire/DPQ_H.csv')

In [128]:
DPQ_2013_2014.head()

SEQN        DPQ010        DPQ020        DPQ030        DPQ040  \
0  73557.0  1.000000e+00  5.397605e-79  5.397605e-79  5.397605e-79   
1  73558.0  2.000000e+00  5.397605e-79  5.397605e-79  5.397605e-79   
2  73559.0  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   
3  73561.0  2.000000e+00  1.000000e+00  5.397605e-79  3.000000e+00   
4  73562.0  3.000000e+00  3.000000e+00  3.000000e+00  3.000000e+00   

         DPQ050        DPQ060        DPQ070        DPQ080        DPQ090  \
0  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   
1  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   
2  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   
3  3.000000e+00  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   
4  3.000000e+00  1.000000e+00  2.000000e+00  1.000000e+00  5.397605e-79   

         DPQ100  
0  1.000000e+00  
1  5.397605e-79  
2           NaN  
3  1.000000e+00  
4  3.000000e+00

In [129]:
# drop final follow-up question
DPQ_2013_2014.drop(['DPQ100'], axis=1, inplace=True)

In [130]:
# values less than 1 are zero
def set_zero(df):
    df[df<1] = 0
    
set_zero(DPQ_2013_2014)

In [131]:
# need to drop rows where not all col are complete
DPQ_2013_2014.dropna(axis=0, how='any', thresh=None, subset=None, inplace=True)

In [132]:
# take a sum across rows and save as new col
DPQ_2013_2014['sum'] = DPQ_2013_2014.iloc[:,1:].sum(axis=1)

In [133]:
# create new target label 'depressed' if sum is greater than 10 --> 1, otherwise 0
DPQ_2013_2014.loc[DPQ_2013_2014['sum'] > 10, 'depressed'] = 1
DPQ_2013_2014.loc[DPQ_2013_2014['sum'] <= 10, 'depressed'] = 0

In [134]:
DPQ_2013_2014.head()

SEQN  DPQ010  DPQ020  DPQ030  DPQ040  DPQ050  DPQ060  DPQ070  DPQ080  \
0  73557.0     1.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0   
1  73558.0     2.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0   
2  73559.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0   
3  73561.0     2.0     1.0     0.0     3.0     3.0     0.0     0.0     0.0   
4  73562.0     3.0     3.0     3.0     3.0     3.0     1.0     2.0     1.0   

   DPQ090   sum  depressed  
0     0.0   1.0        0.0  
1     0.0   2.0        0.0  
2     0.0   0.0        0.0  
3     0.0   9.0        0.0  
4     0.0  19.0        1.0

In [135]:
DPQ_2013_2014.shape

(5393, 12)

In [136]:
DPQ_H = DPQ_2013_2014[['SEQN', 'depressed']]

In [137]:
DPQ_H.head()

SEQN  depressed
0  73557.0        0.0
1  73558.0        0.0
2  73559.0        0.0
3  73561.0        0.0
4  73562.0        1.0

In [138]:
len(DPQ_2013_2014[DPQ_2013_2014['depressed']==1])  # about 8% depressed = true

435

## Demographic Data

### Demographic Variables and Sample Weights (DEMO_H)

In [139]:
# start with demographic data
DEMO_2013_2014 = pd.read_csv('./Demographics/DEMO_H.csv')

In [140]:
DEMO_2013_2014.head(10)

SEQN  SDDSRVYR  RIDSTATR  RIAGENDR      RIDAGEYR  RIDAGEMN  RIDRETH1  \
0  73557.0       8.0       2.0       1.0  6.900000e+01       NaN       4.0   
1  73558.0       8.0       2.0       1.0  5.400000e+01       NaN       3.0   
2  73559.0       8.0       2.0       1.0  7.200000e+01       NaN       3.0   
3  73560.0       8.0       2.0       1.0  9.000000e+00       NaN       3.0   
4  73561.0       8.0       2.0       2.0  7.300000e+01       NaN       3.0   
5  73562.0       8.0       2.0       1.0  5.600000e+01       NaN       1.0   
6  73563.0       8.0       2.0       1.0  5.397605e-79       5.0       3.0   
7  73564.0       8.0       2.0       2.0  6.100000e+01       NaN       3.0   
8  73565.0       8.0       1.0       1.0  4.200000e+01       NaN       2.0   
9  73566.0       8.0       2.0       2.0  5.600000e+01       NaN       3.0   

   RIDRETH3  RIDEXMON  RIDEXAGM  ...  DMDHREDU  DMDHRMAR  DMDHSEDU  \
0       4.0       1.0       NaN  ...       3.0       4.0       NaN   
1       3.0       1.0       NaN  ...       3.0       1.0       1.0   
2       3.0       2.0       NaN  ...       4.0       1.0       3.0   
3       3.0       1.0     119.0  ...       3.0       1.0       4.0   
4       3.0       1.0       NaN  ...       5.0       1.0       5.0   
5       1.0       1.0       NaN  ...       4.0       3.0       NaN   
6       3.0       2.0       6.0  ...       5.0       1.0       5.0   
7       3.0       2.0       NaN  ...       5.0       2.0       NaN   
8       2.0       NaN       NaN  ...       3.0       1.0       5.0   
9       3.0       1.0       NaN  ...       3.0       3.0       NaN   

       WTINT2YR      WTMEC2YR  SDMVPSU  SDMVSTRA  INDHHIN2  INDFMIN2  INDFMPIR  
0  13281.237386  1.348104e+04      1.0     112.0       4.0       4.0      0.84  
1  23682.057386  2.447177e+04      1.0     108.0       7.0       7.0      1.78  
2  57214.803319  5.719329e+04      1.0     109.0      10.0      10.0      4.51  
3  55201.178592  5.576651e+04      2.0     109.0       9.0       9.0      2.52  
4  63709.667069  6.554187e+04      2.0     116.0      15.0      15.0      5.00  
5  24978.144602  2.534499e+04      1.0     111.0       9.0       9.0      4.79  
6  18040.818432  1.791356e+04      1.0     105.0      15.0      15.0      5.00  
7  60325.095254  6.175865e+04      1.0     114.0      10.0      10.0      5.00  
8  23307.675629  5.397605e-79      2.0     106.0      15.0      15.0      5.00  
9  16351.783140  1.748012e+04      1.0     112.0       4.0       4.0      0.48  

[10 rows x 47 columns]

Demographics variables:
- DMDBORN4: Due to the concerns of disclosure risk, starting in 2011, country of birth was recoded into two categories: 1) Born in 50 U.S. states or Washington, DC; and 2) Born in other countries, including U.S. territories.

- DMDEDUC2: This variable is the highest grade or level of education completed by adults 20 years and older. The response categories are: less than 9th grade education, 9-11th grade education (includes 12th grade and no diploma), High school graduate/GED, some college or associates (AA) degree, and college graduate or higher.

- DMDHHSIZ: This variable is the number of people in the participant’s household. The values for this variable range from 1 to 7. Due to disclosure concerns, households that are comprised of 7 or more people are included in the category that is labeled ‘7 or more’.

- RIDAGEYR: Age in years, at the time of the screening interview, is reported for survey participants between the ages of 1 and 79 years of age. All responses of participants aged 80 years and older are coded as ‘80.’

- RIAGENDR: Gender: male 1, female 2

- INDHHIN2: Total household income (reported as a range value in dollars), range 1-15, 77 refused, 99 dont know

- DMQMIL1Z: Served active duty in US Armed Forces, 1 yes, 0 no, 7 refuse, 9 dont know

- DMDMARTL: Marital status - 1 Married, 2 widowed, 3 divorced, 4 separated, 5 never married, 6 living with partner, 77 refused	


In [141]:
# filter only relevant cols
DEMO_H = DEMO_2013_2014.filter(['SEQN','RIDRETH1', 'DMDEDUC2', 'DMDHHSIZ', 'RIDAGEYR', 'RIAGENDR','INDHHIN2', 
                        'DMQMIL1Z', 'DMDMARTL'], axis=1)

In [142]:
# rename col to be more useful
demo_cols = {'SEQN': 'SEQN', 'RIDRETH1':'race_ethnicity', 'DMDEDUC2': 'edu_level', 'DMDHHSIZ': '#_ppl_household', 
               'RIDAGEYR': 'age', 'RIAGENDR': 'gender','INDHHIN2': 'annual_HI', 
                        'DMQMIL1Z': 'served_army', 'DMDMARTL':'marital_status'}

DEMO_H.columns = DEMO_H.columns.map(demo_cols)

In [143]:
DEMO_H.head()

SEQN  race_ethnicity  edu_level  #_ppl_household   age  gender  \
0  73557.0             4.0        3.0              3.0  69.0     1.0   
1  73558.0             3.0        3.0              4.0  54.0     1.0   
2  73559.0             3.0        4.0              2.0  72.0     1.0   
3  73560.0             3.0        NaN              4.0   9.0     1.0   
4  73561.0             3.0        5.0              2.0  73.0     2.0   

   annual_HI  marital_status  
0        4.0             4.0  
1        7.0             1.0  
2       10.0             1.0  
3        9.0             NaN  
4       15.0             1.0

## Dietary Data

### Dietary Interview - Total Nutrient Intakes, First Day (DR1TOT_H)

In [144]:
DIET_2013_2014 = pd.read_csv('./Dietary/DR1TOT_H.csv')

Dietary variables:
- DR1TCAFF : caffeine - Caffeine (mg) range 0 to 2448
- DRQSDIET - On special diet? - 1,2,9, missing
- DR1TPROT - Protein (gm) - 0 to 869.49, missing
- DR1TCARB - Carbohydrate (gm) - 8.67 to 1423.87, missing
- DR1TSUGR - Total sugars (gm) - 0.13 to 1115.5, missing
- DR1TFIBE - Dietary fiber (gm) - 0 to 136.6, missing
- DR1TTFAT - Total fat (gm) - 0 to 548.38, missing
- DR1TCHOL - Cholesterol (mg) - 0 to 3515, missing
- DR1TVB6 - Vitamin B6 (mg)
- DR1TFOLA - Total folate (mcg) - 0 to 3172, missing
- DR1TVB12 - Vitamin B12 (mcg) - 0 to 110, missing
- DR1TVD - Vitamin D (D2 + D3) (mcg) - 0 to 86.3, missing
- DR1TMAGN - Magnesium (mg) - 0 2725, missing
- DR1TIRON - Iron (mg) - 0 to 130, missing
- DR1TZINC - Zinc (mg) - 0 to 116, missing
- DR1TSELE - Selenium (mcg) - 0 to 986.5, missing
- DR1_320Z - Total plain water drank yesterday (gm) - 0 13440, missing

In [145]:
# filter only relevant cols
DIET_H = DIET_2013_2014.filter(['SEQN', 'DR1TCAFF', 'DRQSDIET', 'DR1TPROT','DR1TCARB','DR1TSUGR','DR1TFIBE',
                               'DR1TTFAT','DR1TCHOL','DR1TVB6','DR1TFOLA','DR1TVB12','DR1TVD',
                               'DR1TMAGN','DR1TIRON','DR1TZINC','DR1TSELE','DR1_320Z'], axis=1)

diet_cols = {'SEQN': 'SEQN', 'DR1TCAFF':'caffeine','DRQSDIET': 'special_diet', 'DR1TPROT':'protein',
             'DR1TCARB':'carb','DR1TSUGR':'sugar','DR1TFIBE':'fiber',
            'DR1TTFAT':'fat','DR1TCHOL':'diet_chol','DR1TVB6':'VitB6','DR1TFOLA':'folate',
             'DR1TVB12':'VitB12','DR1TVD':'VitD', 'DR1TMAGN':'magnesium',
             'DR1TIRON':'iron','DR1TZINC':'zinc','DR1TSELE':'selenium','DR1_320Z':'water'}

DIET_H.columns = DIET_H.columns.map(diet_cols)

In [146]:
DIET_H.head()

SEQN      caffeine  special_diet  protein    carb   sugar  fiber  \
0  73557.0  2.030000e+02           2.0    43.63  239.59  176.47   10.8   
1  73558.0  2.400000e+02           2.0   338.13  423.78   44.99   16.7   
2  73559.0  4.500000e+01           1.0    64.61  224.39  102.90    9.9   
3  73560.0  5.397605e-79           2.0    77.75  162.92   80.58   10.6   
4  73561.0  2.400000e+01           1.0    55.24  178.20   87.78   12.3   

      fat  diet_chol  VitB6  folate  VitB12  VitD  magnesium   iron   zinc  \
0   52.81      209.0  0.951   285.0    2.79   3.3      206.0   8.41   8.85   
1  124.29     2584.0  4.415  1243.0   21.45  15.2      903.0  26.88  30.36   
2   65.97       88.0  1.906   423.0    3.78   4.0      210.0  17.57   8.98   
3   58.27      187.0  1.807   275.0    8.76   9.9      183.0  14.19  13.84   
4   55.36       41.0  2.757   390.0    8.30  23.5      396.0  17.72  17.73   

   selenium   water  
0      62.5   960.0  
1     715.8   360.0  
2      98.3  1254.0  
3      91.2   225.0  
4      68.7   240.0

## Questionnaire Data

### Diabetes

variables
- DIQ010 : doc_diabetes - {Have you/Has SP}} ever been told by a doctor or health professional that {you have/{he/she/SP} has} diabetes or sugar diabetes? 1 yes, 2 no, 3 boarderline
- DIQ160 : doc_prediabetes - {Have you/Has SP} ever been told by a doctor or other health professional that {you have/SP has} any of the following: prediabetes, impaired fasting glucose, impaired glucose tolerance, borderline diabetes or that {your/her/his} blood sugar is higher than normal but not high enough to be called diabetes or sugar diabetes? 1 yes, 2 no

In [147]:
DIQ_2013_2014 = pd.read_csv('./Questionnaire/DIQ_H.csv')

In [148]:
DIQ_2013_2014.columns

Index(['SEQN', 'DIQ010', 'DID040', 'DIQ160', 'DIQ170', 'DIQ172', 'DIQ175A',
       'DIQ175B', 'DIQ175C', 'DIQ175D', 'DIQ175E', 'DIQ175F', 'DIQ175G',
       'DIQ175H', 'DIQ175I', 'DIQ175J', 'DIQ175K', 'DIQ175L', 'DIQ175M',
       'DIQ175N', 'DIQ175O', 'DIQ175P', 'DIQ175Q', 'DIQ175R', 'DIQ175S',
       'DIQ175T', 'DIQ175U', 'DIQ175V', 'DIQ175W', 'DIQ175X', 'DIQ180',
       'DIQ050', 'DID060', 'DIQ060U', 'DIQ070', 'DIQ230', 'DIQ240', 'DID250',
       'DID260', 'DIQ260U', 'DIQ275', 'DIQ280', 'DIQ291', 'DIQ300S', 'DIQ300D',
       'DID310S', 'DID310D', 'DID320', 'DID330', 'DID341', 'DID350', 'DIQ350U',
       'DIQ360', 'DIQ080'],
      dtype='object')

In [149]:
DIQ_H = DIQ_2013_2014.filter(['SEQN','DIQ010', 'DIQ160'], axis=1)

In [150]:
# rename col to be more useful
DIQ_cols = {'SEQN':'SEQN', 'DIQ010':'doc_diabetes', 'DIQ160': 'doc_prediabetes'}

DIQ_H.columns = DIQ_H.columns.map(DIQ_cols)

In [151]:
DIQ_H.head()

SEQN  doc_diabetes  doc_prediabetes
0  73557.0           1.0              NaN
1  73558.0           1.0              NaN
2  73559.0           1.0              NaN
3  73560.0           2.0              NaN
4  73561.0           2.0              2.0

### Diet Questionnaire

Variables:
- CBQ505 : fast_food_12m - In the past 12 months, did you buy food from fast food or pizza places?
- DBD895 : #meals_not_home - During the past 7 days, how many meals {did you/did SP} get that were prepared away from home in places such as restaurants, fast food places, food stands, grocery stores, or from vending machines?
    - 0 None, 1 to 21 range, 5555 more than 21 meals per week, 7777 refused, 9999 dont know
- DBD900 : #meals_fast_food - How many of those meals {did you/did SP} get from a fast-food or pizza place?
- DBQ700 : how_healthy_diet - In general, how healthy is {your/his/her} overall diet? 
    - Excellent to Poor: 1-5

In [152]:
DBQ_2013_2014 = pd.read_csv('./Questionnaire/DBQ_H.csv')

In [153]:
DBQ_2013_2014.columns

Index(['SEQN', 'DBQ010', 'DBD030', 'DBD041', 'DBD050', 'DBD055', 'DBD061',
       'DBQ073A', 'DBQ073B', 'DBQ073C', 'DBQ073D', 'DBQ073E', 'DBQ073U',
       'DBQ700', 'DBQ197', 'DBQ223A', 'DBQ223B', 'DBQ223C', 'DBQ223D',
       'DBQ223E', 'DBQ223U', 'DBQ229', 'DBQ235A', 'DBQ235B', 'DBQ235C',
       'DBQ301', 'DBQ330', 'DBQ360', 'DBQ370', 'DBD381', 'DBQ390', 'DBQ400',
       'DBD411', 'DBQ421', 'DBQ424', 'DBD895', 'DBD900', 'DBD905', 'DBD910',
       'CBQ596', 'CBQ606', 'CBQ611', 'CBQ505', 'CBQ535', 'CBQ540', 'CBQ545',
       'CBQ550', 'CBQ552', 'CBQ580', 'CBQ585', 'CBQ590'],
      dtype='object')

In [154]:
DBQ_H = DBQ_2013_2014.filter(['SEQN','DBD900', 'DBQ700'], axis=1)

In [155]:
# rename col to be more useful
DBQ_cols = {'SEQN':'SEQN',
                  'DBD900':'#meals_fast_food', 'DBQ700': 'how_healthy_diet'}

DBQ_H.columns = DBQ_H.columns.map(DBQ_cols)

In [156]:
DBQ_H.head()

SEQN  #meals_fast_food  how_healthy_diet
0  73557.0      8.000000e+00               2.0
1  73558.0               NaN               3.0
2  73559.0      5.397605e-79               3.0
3  73560.0               NaN               NaN
4  73561.0               NaN               2.0

### Sight & Hearing

variables:
- DLQ010 : difficult_hearing - 1 yes, 2 no
- DLQ020 : difficult_seeing - 1 yes, 2 no
- DLQ050 - Have serious difficulty walking ? 
    - 1=yes, 2=no, 7/9
- DLQ080 - Have difficulty doing errands alone ? 
    - 1=yes, 2=no, 7/9


In [157]:
DLQ_2013_2014 = pd.read_csv('./Questionnaire/DLQ_H.csv')

In [158]:
DLQ_2013_2014.columns

Index(['SEQN', 'DLQ010', 'DLQ020', 'DLQ040', 'DLQ050', 'DLQ060', 'DLQ080'], dtype='object')

In [159]:
DLQ_H = DLQ_2013_2014.filter(['SEQN','DLQ010','DLQ020','DLQ050','DLQ080'], axis=1)

In [160]:
# rename col to be more useful
DLQ_cols = {'SEQN':'SEQN', 'DLQ010': 'difficult_hearing', 'DLQ020': 'difficult_seeing',
           'DLQ050':'difficult_walking','DLQ080':'difficult_errands'}

DLQ_H.columns = DLQ_H.columns.map(DLQ_cols)

In [161]:
DLQ_H.head()

SEQN  difficult_hearing  difficult_seeing  difficult_walking  \
0  73557.0                2.0               2.0                2.0   
1  73558.0                2.0               2.0                2.0   
2  73559.0                1.0               2.0                2.0   
3  73560.0                2.0               2.0                2.0   
4  73561.0                2.0               2.0                2.0   

   difficult_errands  
0                2.0  
1                2.0  
2                2.0  
3                NaN  
4                2.0

### Drugs

variables: 
- DUQ211 : used_marijuana - Have you ever smoked marijuana or hashish at least once a month for more than one year?
- DUQ230 : # days used marijuana or hashish/month
- DUQ240 : used_CMH - Have you ever used cocaine, crack cocaine, heroin, or methamphetamine?
- DUQ370 : injected_drugs - Have you ever, even once, used a needle to inject a drug not prescribed by a doctor?

    - 1 yes, 2 no, 7 refused, 9 don't know

In [167]:
DUQ_2013_2014 = pd.read_csv('./Questionnaire/DUQ_H.csv')

In [168]:
DUQ_H = DUQ_2013_2014.filter(['SEQN', 'DUQ240'], axis=1)

In [169]:
DUQ_cols = {'SEQN':'SEQN','DUQ240':'used_CMH'}

DUQ_H.columns = DUQ_H.columns.map(DUQ_cols)

In [170]:
DUQ_H.head()

SEQN  used_CMH
0  73557.0       7.0
1  73558.0       1.0
2  73562.0       2.0
3  73564.0       2.0
4  73566.0       2.0

### Health Care

HSQ_H
- HSD010 - General health condition
1 - Excellent
2 - Very good
3 - Good
4 - Fair
5 - Poor
7 - Refused, 9 - Don’t Know

HUQ_H
- HUQ010 - General health condition
1 - Excellent
2 - Very good
3 - Good
4 - Fair
5 - Poor
7 - Refused, 9 - Don’t Know
- HUQ041 - Type place most often go for healthcare
1- Clinic or health center
2 - Doctor's office or HMO
3 - Hospital emergency room
4 - Hospital outpatient department	
5 - Some other place
6 - Doesn't go to one place most often
77 - Refused, 99 - Don't know
- HUQ051 - #times receive healthcare over past year
0 - 8 are ranges, 77, 99
- HUQ090 - Seen mental health professional/past yr
1, 2, 7, 9


In [175]:
HUQ_2013_2014 = pd.read_csv('./Questionnaire/HUQ_H.csv')

HUQ_H = HUQ_2013_2014.filter(['SEQN', 'HUQ010', 'HUQ041','HUQ051'], axis=1)
HUQ_cols = {'SEQN':'SEQN', 'HUQ010': 'gen_health_cond','HUQ041':'place_healthcare',
            'HUQ051':'healthcare_past_yr'}

HUQ_H.columns = HUQ_H.columns.map(HUQ_cols)

In [176]:
HUQ_H.head()

SEQN  gen_health_cond  place_healthcare  healthcare_past_yr
0  73557.0              2.0               2.0                 5.0
1  73558.0              4.0               2.0                 5.0
2  73559.0              3.0               2.0                 2.0
3  73560.0              2.0               NaN                 1.0
4  73561.0              5.0               2.0                 4.0

### Health Insurance

variables:
- HIQ011 : health_insurance - {Are you/Is SP} covered by health insurance or some other kind of health care plan? 1 yes, 2 no

In [177]:
HIQ_2013_2014 = pd.read_csv('./Questionnaire/HIQ_H.csv')

HIQ_H = HIQ_2013_2014.filter(['SEQN', 'HIQ011'], axis=1)
HIQ_cols = {'SEQN':'SEQN', 'HIQ011': 'health_insurance'}

HIQ_H.columns = HIQ_H.columns.map(HIQ_cols)

In [178]:
HIQ_H.head()

SEQN  health_insurance
0  73557.0               1.0
1  73558.0               2.0
2  73559.0               1.0
3  73560.0               1.0
4  73561.0               1.0

### Househole Characteristics

HOQ_H
- HOD050 - Number of rooms in home
1-13, 777, 999
- HOQ065 - Home owned, bought, rented, other
1 - Owned or being bought
2 - Rented
3 - Other arrangements
7 - Refused, 9 - Don't know


In [179]:
HOQ_2013_2014 = pd.read_csv('./Questionnaire/HOQ_H.csv')

HOQ_H = HOQ_2013_2014.filter(['SEQN', 'HOD050','HOQ065'], axis=1)
HOQ_cols = {'SEQN':'SEQN', 'HOD050': '#_rooms','HOQ065':'home_ownership'}

HOQ_H.columns = HOQ_H.columns.map(HOQ_cols)

In [180]:
HOQ_H.head()

SEQN  #_rooms  home_ownership
0  73557.0      4.0             2.0
1  73558.0      7.0             1.0
2  73559.0      6.0             1.0
3  73560.0      5.0             1.0
4  73561.0      8.0             1.0

### Early Childhood

ECQ_H
- ECQ020 - Mother smoked when pregnant
1=yes, 2=no, 7/9

In [181]:
ECQ_2013_2014 = pd.read_csv('./Questionnaire/ECQ_H.csv')

ECQ_H = ECQ_2013_2014.filter(['SEQN', 'ECQ020'], axis=1)
ECQ_cols = {'SEQN':'SEQN', 'ECQ020': 'mom_smoked_preg'}

ECQ_H.columns = ECQ_H.columns.map(ECQ_cols)

In [182]:
ECQ_H.head()

SEQN  mom_smoked_preg
0  73560.0              2.0
1  73563.0              2.0
2  73569.0              2.0
3  73570.0              2.0
4  73572.0              2.0

### Food Security

FSQ_H
- FSD032A - HH Worried run out of food
    - 1 - Often true
    - 2 - Sometimes true
    -  - Never true
    - 7 - Refused, 9 - Don't know
- FSD032B - HH Food didn't last
    - “ “
- FSD032C - HH Couldn't afford balanced meals
    - “ “
- FSD041 - HH Adults cut size or skip meals
    - 1, 2, 7, 9
- FSD032D - HH Relied on low-cost food for child
1 - Often true
2 - Sometimes true
3 - Never true
7 - Refused, 9 - Don't know
- FSDHH - Household food security category
1 - HH full food security: 0
2 - HH marginal food security: 1-2
3 - HH low food security: 3-5 (HH w/o child) / 3-7 (HH w/ child)
4 - HH very low food security: 6-10 (HH w/o child) / 8-18 (HH w/ child)
No 7/9, just missing
- FSQ012 - HH FS benefit: receive in last 12 months
1, 2, 7, 9
- FSQ162 - HH WIC benefit: receive in last 12 month
1,2,7,9


In [183]:
FSQ_2013_2014 = pd.read_csv('./Questionnaire/FSQ_H.csv')

FSQ_H = FSQ_2013_2014.filter(['SEQN', 'FSD032A', 'FSD032B', 'FSD032C', 'FSD041', 'FSD032D', 'FSDHH',
                             'FSQ012', 'FSQ162'], axis=1)

 
FSQ_cols = {'SEQN': 'SEQN', 'FSD032A':'run_out_food', 'FSD032B':'food_didnt_last', 'FSD032C':'afford_balanced_meals', 
             'FSD041':'adults_skip_meals', 'FSD032D':'relied_cheap_food', 'FSDHH':'food_security_cat',
             'FSQ012':'food_stamps', 'FSQ162':'WIC'}

FSQ_H.columns = FSQ_H.columns.map(FSQ_cols)

In [184]:
FSQ_H.head()

SEQN  run_out_food  food_didnt_last  afford_balanced_meals  \
0  73557.0           3.0              3.0                    3.0   
1  73558.0           3.0              3.0                    3.0   
2  73559.0           3.0              3.0                    3.0   
3  73560.0           3.0              3.0                    3.0   
4  73561.0           3.0              3.0                    3.0   

   adults_skip_meals  relied_cheap_food  food_security_cat  food_stamps  WIC  
0                NaN                NaN                1.0          1.0  NaN  
1                NaN                3.0                1.0          1.0  2.0  
2                NaN                NaN                1.0          NaN  NaN  
3                NaN                3.0                1.0          NaN  2.0  
4                NaN                NaN                1.0          NaN  NaN

### Income

INQ_H
- IND235 - Monthly family income
    - 1-12 ranges, 77, 99
- INQ244 - Family has savings more than $5000
    - 1,2,7,9


In [185]:
INQ_2013_2014 = pd.read_csv('./Questionnaire/INQ_H.csv')

INQ_H = INQ_2013_2014.filter(['SEQN', 'IND235', 'INQ244'], axis=1)

INQ_cols = {'SEQN': 'SEQN', 'IND235':'monthly_income', 'INQ244':'savings_5000'}

INQ_H.columns = INQ_H.columns.map(INQ_cols)


In [186]:
INQ_H.head()

SEQN  monthly_income  savings_5000
0  73557.0             4.0           9.0
1  73558.0             5.0           1.0
2  73559.0            10.0           NaN
3  73560.0             9.0           NaN
4  73561.0            11.0           NaN

### Medical Conditions

variables:
- MCQ010 : doc_asthma - Has a doctor or other health professional ever told {you/SP} that {you have/s/he/SP has} asthma (az-ma)?
- MCQ080 : doc_overweight - Has a doctor or other health professional ever told {you/SP} that {you were/s/he/SP was} overweight?
- MCQ160a :  doc_arthritis - Has a doctor or other health professional ever told {you/SP} that {you/s/he} had arthritis (ar-thry-tis)?
- MCQ160b :  doc_CHF - Has a doctor or other health professional ever told {you/SP} that {you/s/he}had congestive heart failure?
- MCQ160c : doc_CHD - Has a doctor or other health professional ever told {you/SP} that {you/s/he} had coronary (kor-o-nare-ee) heart disease?
- MCQ160e : doc_heart_attack - Has a doctor or other health professional ever told {you/SP} that {you/s/he} had a heart attack (also called myocardial infarction (my-o-car-dee-al in-fark-shun))?
- MCQ160f : doc_stroke - Has a doctor or other health professional ever told {you/SP} that {you/s/her} had a stroke?
- MCQ160k : doc_chronic_bronchitis - Has a doctor or other health professional ever told {you/SP} that {you/s/he} . . .had chronic bronchitis?
- MCQ160l : doc_liver_condition - Has a doctor or other health professional ever told {you/SP} that {you/s/he} . . .had any kind of liver condition?
- MCQ160m : doc_thyroid_problem - Has a doctor or other health professional ever told {you/SP} that {you/s/he} had another thyroid (thigh-roid) problem?
- MCQ160o : doc_COPD - Has a doctor or other health professional ever told {you/SP} that {you/s/he} had COPD?
- MCQ220 : doc_cancer - {Have you/Has SP} ever been told by a doctor or other health professional that {you/s/he} had cancer or a malignancy (ma-lig-nan-see) of any kind?
- MCQ365a : doc_lose_weight - To lower {your/SP's} risk for certain diseases, during the past 12 months {have you/has s/he} ever been told by a doctor or health professional to: control {your/his/her} weight or lose weight?
- MCQ365b : doc_exercise - To lower {your/SP's} risk for certain diseases, during the past 12 months {have you/has s/he} ever been told by a doctor or health professional to: increase {your/his/her} physical activity or exercise?
- MCQ053 - Taking treatment for anemia/past 3 mos
    - 1,2,7,9
- MCQ084 - Difficulties in thinking or remembering
    - 1, 2, 7, 9
- MCQ230a - What kind of cancer
    - 10-39 different kinds, 66 = more than 3, 77, 99


In [187]:
MCQ_2013_2014 = pd.read_csv('./Questionnaire/MCQ_H.csv')

MCQ_H = MCQ_2013_2014.filter(['SEQN', 'MCQ010', 'MCQ080', 'MCQ160A', 'MCQ160B', 'MCQ160C', 'MCQ160E',
 'MCQ160F', 'MCQ160K', 'MCQ160L', 'MCQ160M', 'MCQ160O', 'MCQ220','MCQ053',
                             'MCQ084','MCQ230a'], axis=1)

 
MCQ_cols = {'SEQN': 'SEQN', 'MCQ010':'doc_asthma', 'MCQ080':'doc_overweight', 'MCQ160A':'doc_arthritis', 
 'MCQ160B':'doc_CHF', 'MCQ160C':'doc_CHD', 'MCQ160E':'doc_heart_attack',
 'MCQ160F':'doc_stroke', 'MCQ160K':'doc_chronic_bronchitis', 'MCQ160L':'doc_liver_condition', 
 'MCQ160M':'doc_thyroid_problem', 'MCQ160O':'doc_COPD', 'MCQ220':'doc_cancer','MCQ053': 'anemia',
                             'MCQ084':'difficult_remembering','MCQ230a':'cancer_type'}

MCQ_H.columns = MCQ_H.columns.map(MCQ_cols)

In [189]:
MCQ_H.head()

SEQN  doc_asthma  doc_overweight  doc_arthritis  doc_CHF  doc_CHD  \
0  73557.0         2.0             1.0            1.0      2.0      2.0   
1  73558.0         1.0             2.0            2.0      2.0      2.0   
2  73559.0         2.0             2.0            2.0      2.0      2.0   
3  73560.0         2.0             NaN            NaN      NaN      NaN   
4  73561.0         2.0             2.0            1.0      2.0      2.0   

   doc_heart_attack  doc_stroke  doc_chronic_bronchitis  doc_liver_condition  \
0               2.0         1.0                     2.0                  2.0   
1               2.0         2.0                     2.0                  2.0   
2               2.0         2.0                     2.0                  2.0   
3               NaN         NaN                     NaN                  NaN   
4               2.0         2.0                     1.0                  2.0   

   doc_thyroid_problem  doc_COPD  doc_cancer  anemia  difficult_remembering  
0                  2.0       2.0         2.0     2.0                    2.0  
1                  2.0       2.0         2.0     2.0                    NaN  
2                  2.0       2.0         1.0     2.0                    2.0  
3                  NaN       NaN         NaN     2.0                    NaN  
4                  1.0       2.0         2.0     2.0                    2.0

In [190]:
len(MCQ_H.columns)

15

### Kidney Conditions - Urology (KIQ_U_H)

variables
- KIQ022: doc_kidney - {Have you/Has SP} ever been told by a doctor or other health professional that {you/s/he} had weak or failing kidneys? Do not include kidney stones, bladder infections, or incontinence.

In [191]:
KIQ_U_2013_2014 = pd.read_csv('./Questionnaire/KIQ_U_H.csv')

KIQ_U_H = KIQ_U_2013_2014.filter(['SEQN', 'KIQ022'], axis=1)
KIQ_U_cols = {'SEQN':'SEQN', 'KIQ022': 'doc_kidney'}

KIQ_U_H.columns = KIQ_U_H.columns.map(KIQ_U_cols)

In [192]:
KIQ_U_H.head()

SEQN  doc_kidney
0  73557.0         2.0
1  73558.0         2.0
2  73559.0         1.0
3  73561.0         1.0
4  73562.0         2.0

### Occupation

OCQ_H
- OCD150 - Type of work done last week
1 - Working at a job or business
2 - With a job or business but not at work
3 - Looking for work, or
4 - Not working at a job or business?
7 - Refused, 9-  Don’t know
- OCQ180 - Hours worked last week at all jobs
    - Range 0-120 hours
    - 77777, 999999


In [193]:
OCQ_2013_2014 = pd.read_csv('./Questionnaire/OCQ_H.csv')

OCQ_H = OCQ_2013_2014.filter(['SEQN', 'OCD150','OCQ180'], axis=1)
OCQ_cols = {'SEQN':'SEQN', 'OCD150': 'work_type','OCQ180':'hrs_worked_week'}

OCQ_H.columns = OCQ_H.columns.map(OCQ_cols)

In [194]:
OCQ_H.head()

SEQN  work_type  hrs_worked_week
0  73557.0        4.0              NaN
1  73558.0        1.0             50.0
2  73559.0        4.0              NaN
3  73561.0        4.0              NaN
4  73562.0        1.0             56.0

### Oral Health

OHQ_H
- OHQ680 - Last yr embarrassed because of mouth
    - 1 very often to 5 never, 7, 9


In [195]:
OHQ_2013_2014 = pd.read_csv('./Questionnaire/OHQ_H.csv')

OHQ_H = OHQ_2013_2014.filter(['SEQN', 'OHQ680'], axis=1)
OHQ_cols = {'SEQN':'SEQN', 'OHQ680': 'embarassed_mouth'}

OHQ_H.columns = OHQ_H.columns.map(OHQ_cols)

In [196]:
OHQ_H.head()

SEQN  embarassed_mouth
0  73557.0               5.0
1  73558.0               5.0
2  73559.0               5.0
3  73560.0               NaN
4  73561.0               5.0

### Osteoperosis

variables:
- OSQ010A : broken_hip - Has a doctor ever told {you/SP} that {you/s/he} had broken or fractured {your/his/her} . . .hip?
- OSQ060 : doc_osteoporosis - Has a doctor ever told {you/SP} that {you/s/he} had osteoporosis, sometimes called thin or brittle bones?

    - 1 yes, 2 no, 7 refused, 9 don't know

In [197]:
OSQ_2013_2014 = pd.read_csv('./Questionnaire/OSQ_H.csv')

OSQ_H = OSQ_2013_2014.filter(['SEQN', 'OSQ010A', 'OSQ060'], axis=1)
OSQ_cols = {'SEQN':'SEQN', 'OSQ010A': 'broken_hip', 'OSQ060':'doc_osteoporosis'}

OSQ_H.columns = OSQ_H.columns.map(OSQ_cols)

In [198]:
OSQ_2013_2014.columns

Index(['SEQN', 'OSQ010A', 'OSQ010B', 'OSQ010C', 'OSQ020A', 'OSQ020B',
       'OSQ020C', 'OSD030AA', 'OSQ040AA', 'OSD050AA', 'OSD030AB', 'OSQ040AB',
       'OSD050AB', 'OSD030AC', 'OSQ040AC', 'OSD050AC', 'OSD030BA', 'OSQ040BA',
       'OSD050BA', 'OSD030BB', 'OSQ040BB', 'OSD050BB', 'OSD030BC', 'OSQ040BC',
       'OSD050BC', 'OSD030BD', 'OSQ040BD', 'OSD050BD', 'OSD030BE', 'OSQ040BE',
       'OSD030BF', 'OSQ040BF', 'OSD030BG', 'OSQ040BG', 'OSD030BH', 'OSQ040BH',
       'OSD030BI', 'OSQ040BI', 'OSD030BJ', 'OSQ040BJ', 'OSD030CA', 'OSQ040CA',
       'OSD050CA', 'OSD030CB', 'OSQ040CB', 'OSD050CB', 'OSD030CC', 'OSQ040CC',
       'OSQ080', 'OSQ090A', 'OSQ100A', 'OSD110A', 'OSQ120A', 'OSQ090B',
       'OSQ100B', 'OSD110B', 'OSQ120B', 'OSQ090C', 'OSQ100C', 'OSD110C',
       'OSQ120C', 'OSQ090D', 'OSQ100D', 'OSD110D', 'OSQ120D', 'OSQ090E',
       'OSQ100E', 'OSD110E', 'OSQ120E', 'OSQ090F', 'OSQ120F', 'OSQ090G',
       'OSQ100G', 'OSD110G', 'OSQ120G', 'OSQ090H', 'OSQ120H', 'OSQ060',
       'OSQ072'

In [199]:
OSQ_H.head()

SEQN  broken_hip  doc_osteoporosis
0  73557.0         2.0               2.0
1  73558.0         2.0               2.0
2  73559.0         2.0               1.0
3  73561.0         2.0               2.0
4  73562.0         2.0               2.0

### Physical Activity

variables:
- PAQ650 - Vigorous recreational activities - In a typical week {do you/does SP} do any vigorous-intensity sports, fitness, or recreational activities that cause large increases in breathing or heart rate like running or basketball for at least 10 minutes continuously? 1 yes 2 no
    - PAD200 - Vigorous activity over past 30 days ^^
- PAQ665 - Moderate recreational activities
    - PAD320 - Moderate activity over past 30 days ^^
- PAD680 : min_sedetary - The following question is about sitting at school, at home, getting to and from places, or with friends including time spent sitting at a desk, traveling in a car or bus, reading, playing cards, watching television, or using a computer. Do not include time spent sleeping. How much time {do you/does SP} usually spend sitting on a typical day?
- PAQ706 - Days physically active at least 60 min.
    - 1-7, 77 99
- PAQ710 - Hours watch TV or videos past 30 days
    - 1-5, 8=don’t watch, 77, 99
- PAQ715 - Hours use computer past 30 days (on average/day)
    - “ “
- PAQ722 - Any physical activity in past 7 days

In [200]:
PAQ_2013_2014 = pd.read_csv('./Questionnaire/PAQ_H.csv')

PAQ_H = PAQ_2013_2014.filter(['SEQN', 'PAQ650', 'PAD200', 'PAQ665', 'PAD320',
           'PAD680', 'PAQ706', 'PAQ710','PAQ715','PAQ722'], axis=1)

PAQ_cols = {'SEQN':'SEQN', 'PAQ650':'vigorous_activity', 'PAD200':'vigorous_activity', 'PAQ665':'moderate_activity',
           'PAD3205':'moderate_activity', 'PAD680':'min_sedetary', 'PAQ706':'days_60min_active', 
            'PAQ710':'hours_TV','PAQ715':'hours_computer','PAQ722':'activity_week'}

PAQ_H.columns = PAQ_H.columns.map(PAQ_cols)

In [201]:
PAQ_H.columns

Index(['SEQN', 'vigorous_activity', 'moderate_activity', 'min_sedetary',
       'days_60min_active', 'hours_TV', 'hours_computer', 'activity_week'],
      dtype='object')

In [202]:
PAQ_H.head()

SEQN  vigorous_activity  moderate_activity  min_sedetary  \
0  73557.0                2.0                2.0         600.0   
1  73558.0                2.0                2.0         540.0   
2  73559.0                2.0                1.0         300.0   
3  73560.0                NaN                NaN           NaN   
4  73561.0                2.0                2.0         480.0   

   days_60min_active  hours_TV  hours_computer  activity_week  
0                NaN       2.0    8.000000e+00            NaN  
1                NaN       4.0    8.000000e+00            NaN  
2                NaN       4.0    5.397605e-79            NaN  
3                5.0       3.0    2.000000e+00            1.0  
4                NaN       1.0    1.000000e+00            NaN

### Physical Functioning

PFQ_H
- PFQ049 - Limitations keeping you from working
    - 1,2,7,9
- PFQ059 - Physical, mental, emotional limitations
    - 1,2,7,9
- PFQ061A - Managing money difficulty
    - 1 no difficulty to 4 unable to do, 5 don’t do it, 7, 9

Many other activities I could include

In [205]:
PFQ_2013_2014 = pd.read_csv('./Questionnaire/PFQ_H.csv')

PFQ_H = PFQ_2013_2014.filter(['SEQN', 'PFQ049', 'PFQ059', 'PFQ061A'], axis=1)

PFQ_cols = {'SEQN':'SEQN', 'PFQ049':'limit_work', 'PFQ059':'PME_limit', 'PFQ061A':'manage_$_difficult'}

PFQ_H.columns = PFQ_H.columns.map(PFQ_cols)

In [206]:
PFQ_H.head()

SEQN  limit_work  PME_limit  manage_$_difficult
0  73557.0         2.0        2.0                 1.0
1  73558.0         2.0        2.0                 NaN
2  73559.0         2.0        2.0                 5.0
3  73560.0         NaN        NaN                 NaN
4  73561.0         2.0        NaN                 1.0

### Sleeping Disorders

variables:
- SLQ060 : doc_sleeping_disorder - {Have you/Has SP} ever been told by a doctor or other health professional that {you have/s/he has} a sleep disorder?
- SLD010H - How much sleep do you get (hours)?
    - 2-12, 77, 99

In [207]:
SLQ_2013_2014 = pd.read_csv('./Questionnaire/SLQ_H.csv')

SLQ_H = SLQ_2013_2014.filter(['SEQN', 'SLQ060','SLD010H'], axis=1)
SLQ_cols = {'SEQN':'SEQN', 'SLQ060': 'doc_sleeping_disorder','SLD010H':'hrs_sleep'}

SLQ_H.columns = SLQ_H.columns.map(SLQ_cols)

In [208]:
SLQ_2013_2014.columns

Index(['SEQN', 'SLD010H', 'SLQ050', 'SLQ060'], dtype='object')

In [209]:
SLQ_H.head()

SEQN  doc_sleeping_disorder  hrs_sleep
0  73557.0                    2.0        7.0
1  73558.0                    2.0        9.0
2  73559.0                    2.0        8.0
3  73561.0                    2.0        9.0
4  73562.0                    1.0        5.0

### Smoking - Cigarette Use (SMQ_H)

variables:
- SMQ020 : smoker - These next questions are about cigarette smoking and other tobacco use. {Have you/Has SP} smoked at least 100 cigarettes in {your/his/her} entire life?

In [210]:
SMQ_2013_2014 = pd.read_csv('./Questionnaire/SMQ_H.csv')

SMQ_H = SMQ_2013_2014.filter(['SEQN', 'SMQ020'], axis=1)
SMQ_cols = {'SEQN':'SEQN', 'SMQ020': 'smoker'}

SMQ_H.columns = SMQ_H.columns.map(SMQ_cols)

In [211]:
SMQ_H.head()

SEQN  smoker
0  73557.0     1.0
1  73558.0     1.0
2  73559.0     1.0
3  73561.0     2.0
4  73562.0     1.0

In [212]:
SMQRTU_2013_2014 = pd.read_csv('./Questionnaire/SMQRTU_H.csv')

SMQRTU_H = SMQRTU_2013_2014.filter(['SEQN', 'SMQ681'], axis=1)
SMQ_cols = {'SEQN':'SEQN', 'SMQ681': 'smoked_5days'}

SMQRTU_H.columns = SMQRTU_H.columns.map(SMQ_cols)

In [213]:
SMQRTU_H.head()

SEQN  smoked_5days
0  73557.0           1.0
1  73558.0           1.0
2  73559.0           2.0
3  73561.0           2.0
4  73562.0           2.0

### Sexual Behavior (SXQ_H)

variables:
- SXQ292 : sexual_orientation_M - Do you think of yourself as...1 Heterosexual or straight (attracted to women), 2 Homosexual or gay (attracted to men), 3 Bisexual (attracted to men and women), 4 Something else, 5 Not sure
- SXQ294 : sexual_orientation_F - ""
- SXQ753 : doc_HPV - Has a doctor or other health care professional ever told you that you had human papillomavirus or HPV? 1 yes, 2 no, 7 refuse, 9 don't know
- SXD171 : lifetime_female_partners - In your lifetime, with how many women have you had any kind of sex?
- SXD101 : lifetime_male_partners - In your lifetime, with how many men have you had any kind of sex?


In [214]:
SXQ_2013_2014 = pd.read_csv('./Questionnaire/SXQ_H.csv')

SXQ_H = SXQ_2013_2014.filter(['SEQN', 'SXQ292', 'SXQ294', 'SXD171', 'SXD101', 'SXQ753'], axis=1)
SXQ_cols = {'SEQN':'SEQN', 'SXQ292' : 'sexual_orientation_M', 'SXQ294' : 'sexual_orientation_F', 
           'SXD171' : 'lifetime_female_partners', 'SXD101' : 'lifetime_male_partners', 'SXQ753' : 'doc_HPV'}

SXQ_H.columns = SXQ_H.columns.map(SXQ_cols)


In [215]:
SXQ_2013_2014.columns

Index(['SEQN', 'SXD021', 'SXQ800', 'SXQ803', 'SXQ806', 'SXQ809', 'SXQ700',
       'SXQ703', 'SXQ706', 'SXQ709', 'SXD031', 'SXD171', 'SXD510', 'SXQ824',
       'SXQ827', 'SXD633', 'SXQ636', 'SXQ639', 'SXD642', 'SXQ410', 'SXQ550',
       'SXQ836', 'SXQ841', 'SXQ853', 'SXD621', 'SXQ624', 'SXQ627', 'SXD630',
       'SXQ645', 'SXQ648', 'SXQ610', 'SXQ251', 'SXQ590', 'SXQ600', 'SXD101',
       'SXD450', 'SXQ724', 'SXQ727', 'SXQ130', 'SXQ490', 'SXQ741', 'SXQ753',
       'SXQ260', 'SXQ265', 'SXQ267', 'SXQ270', 'SXQ272', 'SXQ280', 'SXQ292',
       'SXQ294'],
      dtype='object')

In [216]:
SXQ_H.head()

SEQN  sexual_orientation_M  sexual_orientation_F  \
0  73557.0                   NaN                   NaN   
1  73558.0                   1.0                   NaN   
2  73562.0                   1.0                   NaN   
3  73564.0                   NaN                   NaN   
4  73566.0                   NaN                   1.0   

   lifetime_female_partners  lifetime_male_partners  doc_HPV  
0                     100.0                     NaN      NaN  
1                      10.0                     NaN      NaN  
2                      12.0                     NaN      NaN  
3                       NaN                     2.0      NaN  
4                       NaN                     4.0      2.0

notes: may want to concat the # of partners and sexual orientation despite gender

### Taste & Smell

CSQ_H
- CSQ010 - Had problem with smell past 12 months?
    - 1,2,7,9
- CSQ080 - Had problem with taste past 12 months?
    - 1,2,7,9
- CSQ200 - Persistent cold/flu last 12 months?
    - “ “
- CSQ204 - Frequent nasal congestion in past 12 mth

In [217]:
CSQ_2013_2014 = pd.read_csv('./Questionnaire/CSQ_H.csv')

CSQ_H = CSQ_2013_2014.filter(['SEQN', 'CSQ010','CSQ080','CSQ200','CSQ204'], axis=1)
CSQ_cols = {'SEQN':'SEQN', 'CSQ010': 'prob_smell','CSQ080':'prob_tatse',
            'CSQ200':'persistent_cold','CSQ204':'nasal_congestion'}

CSQ_H.columns = CSQ_H.columns.map(CSQ_cols)

In [218]:
CSQ_H.head()

SEQN  prob_smell  prob_tatse  persistent_cold  nasal_congestion
0  73557.0         2.0         2.0              2.0               2.0
1  73558.0         2.0         2.0              2.0               2.0
2  73559.0         2.0         2.0              2.0               2.0
3  73561.0         2.0         2.0              2.0               2.0
4  73562.0         1.0         1.0              2.0               2.0

### Alcohol Use (ALQ_H)

Variables: 
- ALQ151 : alcoholic - Was there ever a time or times in {your/SP's} life when {you/he/she} drank {DISPLAY NUMBER} or more drinks of any kind of alcoholic beverage almost every day? 1 yes, 2 no

In [219]:
ALQ_2013_2014 = pd.read_csv('./Questionnaire/ALQ_H.csv')

ALQ_H = ALQ_2013_2014.filter(['SEQN', 'ALQ151'], axis=1)
ALQ_cols = {'SEQN':'SEQN', 'ALQ151': 'alcoholic'}

ALQ_H.columns = ALQ_H.columns.map(ALQ_cols)

In [220]:
ALQ_H.head()

SEQN  alcoholic
0  73557.0        1.0
1  73558.0        1.0
2  73559.0        2.0
3  73561.0        2.0
4  73562.0        2.0

### Weight History

WHQ_H
- WHQ030 - How do you consider your weight
    - 1 - Overweight
    - 2 - Underweight,
    - 3 - About the right weight?
    - 7 - Refused, 9 - Don’t Know
- WHQ070 - Tried to lose weight in past year
    - 1,2,7,9


In [221]:
WHQ_2013_2014 = pd.read_csv('./Questionnaire/WHQ_H.csv')

WHQ_H = WHQ_2013_2014.filter(['SEQN', 'WHQ030','WHQ070'], axis=1)
WHQ_cols = {'SEQN':'SEQN', 'WHQ030':'consider_weight','WHQ070':'tried_lose_weight'}

WHQ_H.columns = WHQ_H.columns.map(WHQ_cols)

In [222]:
WHQ_H.head()

SEQN  consider_weight  tried_lose_weight
0  73557.0              3.0                NaN
1  73558.0              3.0                2.0
2  73559.0              3.0                2.0
3  73561.0              2.0                2.0
4  73562.0              1.0                1.0

## Laboratory Data

### Cholesterol - HDL (HDL_H)

variables:
- LBDHDD : HDL - Direct HDL-Cholesterol (mg/dL), range 10-173

In [223]:
HDL_2013_2014 = pd.read_csv('./Laboratory/HDL_H.csv')

HDL_H = HDL_2013_2014.filter(['SEQN', 'LBDHDD'], axis=1)
HDL_cols = {'SEQN':'SEQN', 'LBDHDD': 'HDL'}

HDL_H.columns = HDL_H.columns.map(HDL_cols)

In [224]:
HDL_H.head()

SEQN   HDL
0  73557.0  65.0
1  73558.0  50.0
2  73559.0  60.0
3  73560.0  61.0
4  73561.0  85.0

### Cholesterol - LDL & Triglycerides (TRIGLY_H)

variables:

- LBDLDL - LBDLDL = ROUND(LBXTC-(LBXHDD+LBXTR/5) for LBXTR less than or equal to 400 mg/dL, and missing for LBXTR greater than 400 mg/dL, range 14-375
- LBXTR : triglyceride - Triglyceride (mg/dL), range 13-4233

In [225]:
TRIGLY_2013_2014 = pd.read_csv('./Laboratory/TRIGLY_H.csv')

TRIGLY_H = TRIGLY_2013_2014.filter(['SEQN', 'LBDLDL', 'LBXTR'], axis=1)
TRIGLY_cols = {'SEQN':'SEQN', 'LBDLDL':'LDL','LBXTR': 'triglyceride'}

TRIGLY_H.columns = TRIGLY_H.columns.map(TRIGLY_cols)

In [226]:
TRIGLY_2013_2014.columns

Index(['SEQN', 'WTSAF2YR', 'LBXTR', 'LBDTRSI', 'LBDLDL', 'LBDLDLSI'], dtype='object')

In [227]:
TRIGLY_H.head()

SEQN    LDL  triglyceride
0  73559.0   56.0          51.0
1  73561.0  101.0          75.0
2  73564.0   97.0          64.0
3  73568.0   67.0          24.0
4  73574.0   75.0          14.0

### Cholesterol - Total (TCHOL_H)

variables:
- LBXTC : tot_cholesterol - Total Cholesterol( mg/dL), range 69-813

In [228]:
TCHOL_2013_2014 = pd.read_csv('./Laboratory/TCHOL_H.csv')

TCHOL_H = TCHOL_2013_2014.filter(['SEQN', 'LBXTC'], axis=1)
TCHOL_cols = {'SEQN':'SEQN', 'LBXTC': 'tot_cholesterol'}

TCHOL_H.columns = TCHOL_H.columns.map(TCHOL_cols)

In [229]:
TCHOL_H.head()

SEQN  tot_cholesterol
0  73557.0            167.0
1  73558.0            170.0
2  73559.0            126.0
3  73560.0            168.0
4  73561.0            201.0

### Glycohemoglobin (GHB_H)

variables:
- LBXGH - Glycohemoglobin (%), range 3.5-17.5

In [230]:
GHB_2013_2014 = pd.read_csv('./Laboratory/GHB_H.csv')

GHB_H = GHB_2013_2014.filter(['SEQN', 'LBXGH'], axis=1)
GHB_cols = {'SEQN':'SEQN', 'LBXGH': 'glycohemoglobin'}

GHB_H.columns = GHB_H.columns.map(GHB_cols)

In [231]:
GHB_H.head()

SEQN  glycohemoglobin
0  73557.0             13.9
1  73558.0              9.1
2  73559.0              8.9
3  73561.0              4.9
4  73562.0              5.5

### Herpes Simplex Virus Type-1 & Type-2 (HSV_H)

variables:
- LBXHE2 : herpes_2 - Herpes Simplex Virus Type 2

In [232]:
HSV_2013_2014 = pd.read_csv('./Laboratory/HSV_H.csv')

HSV_H = HSV_2013_2014.filter(['SEQN', 'LBXHE2'], axis=1)
HSV_cols = {'SEQN':'SEQN', 'LBXHE2': 'herpes_2'}

HSV_H.columns = HSV_H.columns.map(HSV_cols)

In [233]:
HSV_H.head(2)

SEQN  herpes_2
0  73568.0       2.0
1  73574.0       2.0

### HIV Antibody Test (HIV_H)

variables:
- LBDHI : HIV - HIV antibody test result, 1 pos, 2, neg, 3 indeterminant

In [234]:
HIV_2013_2014 = pd.read_csv('./Laboratory/HIV_H.csv')

HIV_H = HIV_2013_2014.filter(['SEQN', 'LBDHI'], axis=1)
HIV_cols = {'SEQN':'SEQN', 'LBDHI': 'HIV'}

HIV_H.columns = HIV_H.columns.map(HIV_cols)

In [235]:
HIV_H.head(2)

SEQN  HIV
0  73558.0  2.0
1  73562.0  2.0

### High-Sensitivity C-Reactive Protein (hs-CRP) (HSCRP_I) - NOT IN 2013-2014

variables:
- LBXHSCRP : CRP - HS C-Reactive Protein (mg/L), values 0-188, missing values are mostly below detection limit --> can be made to 0

In [236]:
# HSCRP_2015_2016 = pd.read_csv('./Laboratory/HSCRP_I.csv')

# HSCRP_I = HSCRP_2015_2016.filter(['SEQN', 'LBXHSCRP '], axis=1)
# HSCRP_cols = {'SEQN':'SEQN', 'LBXHSCRP ': 'CRP'}

# HSCRP_I.columns = HSCRP_I.columns.map(HSCRP_cols)

### Serums

CUSEZN_H
- LBXSSE - Serum Selenium (ug/L) - cont range, missing
- LBXSZN - Serum Zinc (ug/dL) - continuous range, missing
- URXUCR - Urinary creatinine (mg/dL) - continuous range, missing

In [237]:
CUSEZN_2013_2014 = pd.read_csv('./Laboratory/CUSEZN_H.csv')

CUSEZN_H = CUSEZN_2013_2014.filter(['SEQN', 'LBXSSE','LBXSZN','URXUCR'], axis=1)
CUSEZN_cols = {'SEQN':'SEQN', 'LBXSSE': 'serum_selenium', 'LBXSZN':'serum_zinc','URXUCR':'serum_creatinine'}

CUSEZN_H.columns = CUSEZN_H.columns.map(CUSEZN_cols)

In [238]:
CUSEZN_H.head()

SEQN  serum_selenium  serum_zinc  serum_creatinine
0  73560.0           112.2        79.9              76.0
1  73564.0           131.2        81.0             242.0
2  73567.0           114.0        73.2             215.0
3  73583.0             NaN         NaN             151.0
4  73585.0           114.5        89.8             100.0

### Folate

FOLATE_H
- LBDRFO - RBC folate (ng/mL) -  continuous range, missing

FOLFMS_H
- LBDFOT - Serum total folate (ng/mL) -  continuous range, missing

In [239]:
FOLATE_2013_2014 = pd.read_csv('./Laboratory/FOLATE_H.csv')
FOLFMS_2013_2014 = pd.read_csv('./Laboratory/FOLFMS_H.csv')

FOLATE_H = FOLATE_2013_2014.filter(['SEQN', 'LBDRFO'], axis=1)
FOLATE_cols = {'SEQN':'SEQN', 'LBDRFO':'rbc_folate'}
FOLFMS_H = FOLFMS_2013_2014.filter(['SEQN', 'LBDFOT'], axis=1)
FOLFMS_cols = {'SEQN':'SEQN', 'LBDFOT':'serum_folate'}

FOLATE_H.columns = FOLATE_H.columns.map(FOLATE_cols)
FOLFMS_H.columns = FOLFMS_H.columns.map(FOLFMS_cols)

In [240]:
FOLFMS_H.head()

SEQN  serum_folate
0  73557.0         16.60
1  73558.0          8.39
2  73559.0         31.20
3  73560.0         30.00
4  73561.0         39.70

In [241]:
FOLATE_H.head()

SEQN  rbc_folate
0  73557.0       503.0
1  73558.0       259.0
2  73559.0       746.0
3  73560.0       450.0
4  73561.0       746.0

### Fasting Glucose

GLU_H
- LBXGLU - Fasting Glucose (mg/dL) - 51 to 421, missing

In [242]:
GLU_2013_2014 = pd.read_csv('./Laboratory/GLU_H.csv')

GLU_H = GLU_2013_2014.filter(['SEQN', 'LBXGLU'], axis=1)
GLU_cols = {'SEQN':'SEQN', 'LBXGLU': 'fasting_glucose'}

GLU_H.columns = GLU_H.columns.map(GLU_cols)

In [243]:
GLU_H.head()

SEQN  fasting_glucose
0  73559.0            193.0
1  73561.0            107.0
2  73564.0            110.0
3  73568.0             89.0
4  73574.0             84.0

### Vitamins B12 and D

VITB12_H
- LBDB12 - Vitamin B12(pg/mL) - range, missing

VID_H
- LBXVIDMS - 25OHD2+25OHD3 (nmol/L) - range, missing (vitamin D)

In [244]:
VITB12_2013_2014 = pd.read_csv('./Laboratory/VITB12_H.csv')

VITB12_H = VITB12_2013_2014.filter(['SEQN', 'LBDB12'], axis=1)
VITB12_cols = {'SEQN':'SEQN', 'LBDB12': 'lab_VitB12'}

VITB12_H.columns = VITB12_H.columns.map(VITB12_cols)


VID_2013_2014 = pd.read_csv('./Laboratory/VID_H.csv')

VID_H = VID_2013_2014.filter(['SEQN', 'LBXVIDMS'], axis=1)
VID_cols = {'SEQN':'SEQN', 'LBXVIDMS': 'lab_VitD'}

VID_H.columns = VID_H.columns.map(VID_cols)


In [245]:
VITB12_H.head()

SEQN  lab_VitB12
0  73557.0       524.0
1  73558.0       507.0
2  73559.0       732.0
3  73561.0       225.0
4  73562.0       750.0

In [246]:
VID_H.head()

SEQN  lab_VitD
0  73557.0      28.9
1  73558.0      61.9
2  73559.0     126.0
3  73560.0      73.3
4  73561.0     108.0

## Examination Data

### Blood Pressure (BPX_H)

variables:
- BPXSY1 : systolic_BP - Systolic: Blood pres (1st rdg) mm Hg
- BPXDI1 : diastolic_BP - Diastolic: Blood pres (1st rdg) mm Hg
- BPXPLS - 60 sec. pulse (30 sec. pulse * 2) - 0 to 180, missing

In [247]:
BPX_2013_2014 = pd.read_csv('./Examination/BPX_H.csv')

BPX_H = BPX_2013_2014.filter(['SEQN', 'BPXSY1', 'BPXDI1','BPXPLS'], axis=1)
BPX_cols = {'SEQN':'SEQN', 'BPXSY1': 'systolic_BP', 'BPXDI1' : 'diastolic_BP','BPXPLS':'pulse'}

BPX_H.columns = BPX_H.columns.map(BPX_cols)

In [248]:
BPX_H.head(2)

SEQN  systolic_BP  diastolic_BP  pulse
0  73557.0        122.0          72.0   86.0
1  73558.0        156.0          62.0   74.0

### Body Measures (BMX_H)

variables:
- BMXBMI : BMI - Body Mass Index (kg/m**2), range 12.1 to 82.9
- BMXWAIST : waist_C - Waist Circumference (cm), 40.2 to 177.9

In [249]:
BMX_2013_2014 = pd.read_csv('./Examination/BMX_H.csv')

BMX_H = BMX_2013_2014.filter(['SEQN', 'BMXBMI', 'BMXWAIST'], axis=1)
BMX_cols = {'SEQN':'SEQN', 'BMXBMI': 'BMI', 'BMXWAIST' : 'waist_C'}

BMX_H.columns = BMX_H.columns.map(BMX_cols)

In [250]:
BMX_H.head(2)

SEQN   BMI  waist_C
0  73557.0  26.7    100.0
1  73558.0  28.6    107.6

### Dual-Energy X-ray Absorptiometry - Whole Body (DXX_H)

variables:
- DXDTOBMD : bone_mineral_density - Total Bone Mineral Density (g/cm^2), range 0.559 to 1.548

In [251]:
DXX_2013_2014 = pd.read_csv('./Examination/DXX_H.csv')

DXX_H = DXX_2013_2014.filter(['SEQN', 'DXDTOBMD'], axis=1)
DXX_cols = {'SEQN':'SEQN', 'DXDTOBMD': 'bone_mineral_density'}

DXX_H.columns = DXX_H.columns.map(DXX_cols)

In [252]:
DXX_H.head(2)

SEQN  bone_mineral_density
0  73558.0                 1.024
1  73560.0                 0.856

### Oral Health

OHXREF_H
- OHAREC - Overall recommendation for care
    - 1 - See a dentist immediately
    - 2 - See a dentist within the next 2 weeks	
    - 3 - See a dentist at your earliest convenience
    - 4 - Continue your regular routine care
    - Missing


In [253]:
OHXREF_2013_2014 = pd.read_csv('./Examination/OHXREF_H.csv')

OHXREF_H = OHXREF_2013_2014.filter(['SEQN', 'OHAREC'], axis=1)
OHXREF_cols = {'SEQN':'SEQN', 'OHAREC': 'oral_rec'}

OHXREF_H.columns = OHXREF_H.columns.map(OHXREF_cols)

In [254]:
OHXREF_H.head()

SEQN  oral_rec
0  73557.0       4.0
1  73558.0       3.0
2  73559.0       3.0
3  73560.0       4.0
4  73561.0       3.0

# Merging dataframes

In [255]:
from functools import reduce

# list of df
data_frames = [DPQ_H, DEMO_H, DIET_H, DIQ_H, DBQ_H, DLQ_H, DUQ_H, HUQ_H, HIQ_H, 
               HOQ_H, ECQ_H, FSQ_H, INQ_H, MCQ_H, KIQ_U_H, OCQ_H, OHQ_H, OSQ_H, 
               PAQ_H, PFQ_H, SLQ_H, SMQ_H, SMQRTU_H, SXQ_H, CSQ_H, ALQ_H, WHQ_H, 
               HDL_H, TRIGLY_H, TCHOL_H, GHB_H, HSV_H, HIV_H, CUSEZN_H, FOLFMS_H, 
               FOLATE_H, GLU_H, VITB12_H, VID_H, BPX_H, BMX_H, DXX_H, OHXREF_H]

df_2013_2014 = reduce(lambda  left,right: pd.merge(left,right,on=['SEQN'],
                                            how='outer'), data_frames)

In [256]:
df_2013_2014.depressed

0        0.0
1        0.0
2        0.0
3        0.0
4        1.0
        ... 
10170    NaN
10171    NaN
10172    NaN
10173    NaN
10174    NaN
Name: depressed, Length: 10175, dtype: float64

In [257]:
df_2013_2014.dropna(subset=['depressed'], inplace=True)

In [258]:
df_2013_2014.shape

(5393, 120)

In [259]:
df_2013_2014.columns

Index(['SEQN', 'depressed', 'race_ethnicity', 'edu_level', '#_ppl_household',
       'age', 'gender', 'annual_HI', 'marital_status', 'caffeine',
       ...
       'fasting_glucose', 'lab_VitB12', 'lab_VitD', 'systolic_BP',
       'diastolic_BP', 'pulse', 'BMI', 'waist_C', 'bone_mineral_density',
       'oral_rec'],
      dtype='object', length=120)

# Export 2013-2014 data

In [260]:
df_2013_2014.to_csv('/Users/carolinesklaver/Desktop/Capstone/NHANES/data/csv_data/df_2013_2014_ext.csv', index=False)